src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SessionMiddleware.cs

```diff
@@ -31,9 +31,33 @@ public SessionMiddleware(RequestDelegate next, ILogger<SessionMiddleware> logger
     }
 
     public Task InvokeAsync(HttpContextCore context)
-        => context.GetEndpoint()?.Metadata.GetMetadata<SessionAttribute>() is { Behavior: not SessionBehavior.None } metadata
+    {
+        var feature = context.Features.Get<ISessionStateFeature>();
```

Comment:
I would like to just use the ISessionStateFeature here if possible - it will make it easier to test